be/src/storage/lake/txn_log_applier.cpp

```diff
@@ -42,10 +42,10 @@ class PrimaryKeyTxnLogApplier : public TxnLogApplier {
     }
 
     ~PrimaryKeyTxnLogApplier() override {
+        // must release primary index before `handle_failure`, or `handle_failure` will fail
```

Comment:
otherwise `handle_failure` will fail?

Reply:
Yes, it will retry to remove entry in `handle_failure`, if not release first, remove will fail.